src/core/interaction/pickingcontroller.cpp

```diff
@@ -52,29 +52,15 @@ void PickingController::propagateEvent(Event* event, EventPropagator* propagator
     if (!event) return;
     if (!pickingEnabled()) return;
 
-    const auto pickId = [&](size2_t coord) -> size_t {
-        if (auto src = src_.lock()) {
-            const auto value = src->readPixel(coord, LayerType::Picking);
-            if (value.a > 0.0) {
-                return PickingManager::colorToIndex(uvec3(value));
-            }
-        }
-        return 0;
-    };
-
     switch (event->hash()) {
         case MouseEvent::chash(): {
             auto me = static_cast<MouseEvent*>(event);
-            const auto coord =
```

Comment:
@ResearchDaniel Why was this clamping removed? 
This will cause UB in the conversion from dvec2 to uvec2 for numbers <= -1.0

Reply:
The intension was to move this check to the pickId function and also avoid getting pickId on the border when it is actually outside. 
Changing the input type to dvec2 should fix it?

Reply:
yes, did that.